docs/user/build-default-versions.rst

```diff
@@ -56,20 +56,25 @@ setuptools:
   All other projects use the latest version.
 
 mock:
-  ``1.0.1`` (could be removed in the future).
+  ``1.0.1``.
+  Projects created after August 7, 2023 won't install this dependency by default.
```

Comment:
I don't think this will break anything, since it's for new projects, so I'm fine doing it sooner.

Reply:
I targeted this for next month to give us some time to review the PR, make the adjustments, etc and maybe write a  pretty small changelog blog post announcing this so we have something to link people to when they ask.